### PR TITLE
mod: bump version of kvdb submodule

### PR DESCRIPTION
**Pull request by guggero**
PR #5992 was merged without a bump in the kvdb version which results in
an error when trying to pull in lnd 0.14.0-beta as a library in other
projects.

I already pushed the kvdb/v1.2.1 tag, so this just fixes the import in external projects.

To avoid this problem in the future, a new script is added that checks for changes in submodules without a bump in the version.
The new procedure for changing submodules now is:
 1. Create a PR with the changes to a submodule, for example `kvdb`.
 2. In the PR, bump the version of that submodule in the root `go.mod`. That version will not exist as a tag yet. But because of our local `replace` directives, that will not matter to any of the CI scripts as they all just follow the `replace` directive.
 3. Once the PR is merged, all that needs to be done is to push a tag for the new submodule version, pointing to the merge commit of the PR.

### CODE REVIEW

File: scripts/check-submodule-version.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+SUBMODULES="cert clock healthcheck kvdb queue ticker"
```
Comment (Roasbeef) [inline scripts/check-submodule-version.sh:3]: I wonder if there's a way to grab this using Go commands?

Comment (Roasbeef) [reply scripts/check-submodule-version.sh:3]: Prob possible with some templating tricks using `go list -f`....I think this is good enough for now, a slim layer to help prevent issues like this in the future!

**Review by Roasbeef — APPROVED** [review]
LGTM 🥻

File: scripts/check-submodule-version.sh (hunk 1 above)
Comment (guggero) [reply scripts/check-submodule-version.sh]: Ah yes, good idea. Added some `go list` magic.

File: scripts/check-submodule-version.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+SUBMODULES="$(go list -m all | grep $ROOT_MODULE/ | cut -d' ' -f1 | cut -d'/' -f4-)"
+BRANCH=$1
+
+for m in $SUBMODULES; do
+  has_changes=0
+  git diff --stat $BRANCH.. | grep -q " $m/" && has_changes=1
+  
+  if [[ $has_changes -eq 1 ]]; then
+    has_bump=0
+    git diff $BRANCH.. -- go.mod | \
```
Comment (bhandras) [inline scripts/check-submodule-version.sh:18]: Very nice!

**Review by bhandras — APPROVED** [review]
LGTM :1st_place_medal:

**Pull request merged**